bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/RowFilterAdapter.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.hadoop.hbase.filter.RegexStringComparator;
+import org.apache.hadoop.hbase.filter.CompareFilter.CompareOp;
+
+import com.google.bigtable.v1.RowFilter;
+import com.google.cloud.bigtable.hbase.adapters.ReaderExpressionHelper;
+import com.google.protobuf.ByteString;
+
+/**
+ * An adapter for row key filters using comparators and operators.
+ * <p>
+ * Note that regular expression comparators are passed as is. This means some
+ * users may need to modify their queries to adhere to the RE2 syntax.
+ * <p>
+ * Currently only the regular expression operator with the EQUAL operator is
+ * supported.
+ */
+public class RowFilterAdapter implements
+  TypedFilterAdapter<org.apache.hadoop.hbase.filter.RowFilter> {
+
+  @Override
+  public RowFilter adapt(FilterAdapterContext context,
+      org.apache.hadoop.hbase.filter.RowFilter filter) throws IOException {
+    ByteArrayComparable comparator = filter.getComparator();
+    if (comparator instanceof RegexStringComparator) {
+      return adaptRegexStringComparator(filter.getOperator(), (RegexStringComparator) comparator);
+    } else if (comparator instanceof BinaryComparator) {
+      return adaptBinaryComparator(context, filter.getOperator(), (BinaryComparator) comparator);
+    }
+    throw new IllegalStateException(String.format("Cannot adapt comparator %s", filter
+        .getComparator().getClass().getCanonicalName()));
+  }
+
+  private RowFilter
+      adaptRegexStringComparator(CompareOp compareOp, RegexStringComparator comparator) {
+    String pattern = FilterAdapterHelper.extractRegexPattern(comparator);
+    switch (compareOp) {
+    case EQUAL:
+        return RowFilter.newBuilder().setRowKeyRegexFilter(ByteString.copyFromUtf8(pattern)).build();
+    case NO_OP:
+        return FilterAdapterHelper.ACCEPT_ALL_FILTER;
+    case LESS:
+    case LESS_OR_EQUAL:
+    case NOT_EQUAL:
+    case GREATER_OR_EQUAL:
+    case GREATER:
+    default:
+        throw new IllegalStateException(String.format("Cannot adapt regex filter with compare op %s",
+          compareOp));
+    }
+  }
+
+  private RowFilter adaptBinaryComparator(
+      FilterAdapterContext context, CompareOp compareOp, BinaryComparator comparator)
+      throws IOException {
+    switch (compareOp) {
+      case EQUAL:
+          return RowFilter.newBuilder()
+              .setRowKeyRegexFilter(
+                  ByteString.copyFrom(
+                      ReaderExpressionHelper.quoteRegularExpression(comparator.getValue())))
+              .build();
+      case NO_OP:
+          // No-op always passes. Instead of attempting to return null or default instance,
+          // include an always-match filter.
+          return FilterAdapterHelper.ACCEPT_ALL_FILTER;
+        case LESS:
+      case LESS_OR_EQUAL:
+      case NOT_EQUAL:
+      case GREATER_OR_EQUAL:
+      case GREATER:
+      default:
+          throw new IllegalStateException(String.format("Cannot handle unknown compare op %s",
```

Comment:
Not sure what you mean, it definitely is not "unknown compare op". You can do away "unknown", but it doesn't give enough information "Cannot handle compare op %s", does it?